### ISSUE TITLE
[BUG] 2.0.9.2 Safe home is nowhere near center, and z offset is off

### ISSUE THREAD

**Issue by KalixtShawxo** [MarlinFirmware/Marlin#23021]
### Did you test the latest `bugfix-2.0.x` code?

No, but I will test it now!

### Bug Description

upgraded to 2.0.9.2 from .1, same safe home settings:
```cpp
#define Z_SAFE_HOMING

#if ENABLED(Z_SAFE_HOMING)
  #define Z_SAFE_HOMING_X_POINT ((X_BED_SIZE) / 2)    // X point for Z homing
  #define Z_SAFE_HOMING_Y_POINT ((Y_BED_SIZE) / 2)    // Y point for Z homing
#endif
```
but now the hot end stops at X30 Y130 as center, when it should be stopping at X108 Y 117
my printable bed area is 216 x 235. 

also, since upgrading, I cannot take the z offset close enough to the bed. previously my z ofset was -3.55, and it printed fine. now the nozzle is too high and I cannot lower it.  changing the z offset does not move it any further down. 

Going to reflash with the old Firmware from 2.0.9.1 and see what happens

### Bug Timeline

new, with upgrade to 2.0.9.2

### Expected behavior

I expected it to work the same as 2.0.9.1 with same setting put in before being compiled.

### Actual behavior

Safe homing is not going to the expected center (see description above) and z offset is not working

### Version of Marlin Firmware

2.0.9.2

### Printer model

Ender 3 Pro with BL-Touch

### Electronics

stock

### Bed Leveling

UBL Bilinear mesh

### Your Slicer

Cura

### Host Software

OctoPrint

### Additional information & file uploads

[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/7413883/Configuration.zip)

**Comment by KalixtShawxo** [comment]
reflashed with compiled 2.0.9.1, homing went to center of bed for the probe, with nozzle over to the right some as expected, whereas with 2.0.9.2 the nozzle was ar x30 putting the bl touch probe close to the left edge. 
I had my z offset to -3.55 as before, but for some reason now it was way off. I am having to adjust it down near -3.9, which it is allowing me to do, whereas 2.0.9.2 would not allow the z offset to adjust. the numbers changed but the nozzle didn't.

**Comment by thisiskeithb** [comment]
Whenever there are homing or leveling issues, we now ask everyone to follow a standard procedure to gather more information:

- Download Marlin [`bugfix-2.0.x`](https://github.com/MarlinFirmware/Marlin/archive/bugfix-2.0.x.zip) to test with the latest code.
- Enable `DEBUG_LEVELING_FEATURE` and `M114_DETAIL` and re-flash the firmware.
- Connect to your printer from host software such as Cura, Printrun or Repetier Host.
- Send `M502` and `M500` to ensure your Configurations are applied.
- Issue the command `M111 S247` to enable maximum logging.
- Perform a `G28` to do your standard homing procedure.
- Do a `G29` to probe the bed. This will also enable bed leveling.
- Do some of the moves that revealed problems before. Take notes.
- Copy the log output into a **.TXT file** and attach it to your next reply.

Repeat this procedure, if needed, to demonstrate inconsistencies. From these logs we should hopefully get a better idea of what's going on with your machine.

**Comment by evg-zhabotinsky** [comment]
Do you use `M206` to set XY home offsets? If so then maybe the upgrade has nothing to do with the first problem (uncentered safe homing) and #23069 might fix that. Reset offsets with `M206 X0 Y0` to check this possibility.

I have no guesses about Z offset problem, except that it might be separate from safe homing XY positioning.

**Comment by KalixtShawxo** [comment]
No, I have not used M206 anywhere. I copied all my .1 setting in the config files to the .2 config files and then compiled. The problem then existed where it did not before. I reflashed with .1 and it is working as expected. I have not had a chance to try the hot-fix yet. But on a separate related issue, I did buy a Ender 3 Max at the time I posted the problem, and was able to upgrade the firmware on it to 2.0.9.2 without any issue. I used my PRO config files as a template for any custom settings. 

Sent from Yahoo Mail on Android 
 
  On Wed, Nov 3, 2021 at 12:35, Evgeniy ***@***.***> wrote:   


Do you use M206 to set XY home offsets? If so then maybe the upgrade has nothing to do with the first problem (uncentered safe homing) and #23069 might fix that.

I have no guesses about Z offset problem, except that it might be separate from safe homing XY positioning.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by RamazanKara** [comment]
Have the same Z Offset issue. Using 2.0.9.2 the Z Offset started to be way off which did not happen in the previous update. I added myself an automation step to auto probe for the Z Offset and my probes are now off by far. Will further test if i can provide any more debug information

**Comment by SeeingBlue** [comment]
I am having the exact same problem with the 2.0.9.2 z offset. I nearly broke something tonight thinking I was doing something wrong. I even enabled the z-probe-offset-wizard and followed that. It was not changing the height of the nozzle when the print started. Rolling back to the previous version sounds like a viable solution for now.

**Comment by J-charles-C** [comment]
Hello,

I had the same problem with 2.1.2. Installing the latest bugfix revision didn't work better. On the other hand, my values of Z, K or of sensitivity of the sensorless did not evolve. I did a reset of the internal memory and it worked perfectly. Am I in the same situation as you?

best

**Comment by KalixtShawxo** [comment]
Not sure what you mean by reset of the internal memory. When I did the upgrade (after matching the changes to the new firmware to the old) mine would not center on the bed like before and z was off as well. Sorry, I'm kinda a newb.
When I reflashed it to the previous version it was working again. Although since then I've had some other issues to deal with that has stopped my trying to do anything with the updates again. 

Sent from Yahoo Mail on Android 
 
  On Wed, Nov 24, 2021 at 11:04, ***@***.***> wrote:   


Hello,

I had the same problem with 2.1.2. Installing the latest bugfix revision didn't work better. On the other hand, my values of Z, K or of sensitivity of the sensorless did not evolve. I did a reset of the internal memory and it worked perfectly. Am I in the same situation as you?

best

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by J-charles-C** [comment]
Hello,

I'm very bad at English, sorry. I copied exactly the same parameters between 2.0.2 and today's bug fix. This did not improve my home centering problem, I had the Z offset offset and it brought back my old value of K. (I had however written the correct one before compilation and flash) I used the command M502, then M501 and everything is back to normal. If my symtoms were the same as the editor of this post, I'm not sure I share the cause. cordially
